spec/services/allocations/finder_spec.rb

```diff
@@ -106,7 +106,37 @@
       let(:filter_params) { { locations: [from_location.id, other_location.id] } }
 
       it 'returns allocations matching either specified from_location or to_location' do
-        expect(allocation_finder.call).to match_array [allocation, other_allocation]
+        expect(allocation_finder.call).to contain_exactly(allocation, other_allocation)
+      end
+    end
+
+    describe 'by status' do
+      let!(:unfilled_allocation) { create :allocation, :unfilled, from_location: from_location, to_location: to_location }
+      let!(:filled_allocation) { create :allocation, :filled, from_location: from_location, to_location: to_location }
+      let!(:cancelled_allocation) { create :allocation, :cancelled, from_location: from_location, to_location: to_location }
+
+      context 'with matching status' do
+        let(:filter_params) { { status: 'filled' } }
+
+        it 'returns allocations matching status' do
+          expect(allocation_finder.call).to contain_exactly(filled_allocation)
+        end
+      end
+
+      context 'with multiple statuses' do
+        let(:filter_params) { { status: 'unfilled,cancelled' } }
+
+        it 'returns allocations matching status' do
+          expect(allocation_finder.call).to contain_exactly(unfilled_allocation, cancelled_allocation)
+        end
+      end
+
+      context 'with mis-matching status' do
+        let(:filter_params) { { status: 'foo' } }
```

Comment:
can you add a test for nil values as well?

Reply:
why not!